### ISSUE TITLE
discuss about backlash

### ISSUE THREAD

**Issue by walterwissmann** [fra589/grbl-Mega-5X#40]
Hi,

@Schildkroet have done routines for backlash: https://github.com/Schildkroet/GRBL-Advanced

Would it be nice to have also this routines for MEGA 5X ?

BR
Walter

**Comment by fra589** [comment]
Hi Walter,
I think it is a very good idea :)
Thi is added to my TODO list... But the list is long.
@++;
Gauthier.

**Comment by jgalvarezr** [comment]
Hi...
I have been pending of this case waiting for its solution. My CNC has a lot of inertia and fails at high speeds.

I understand that this is altruistic work, but I would like to know if there will be a solution.

Thanks for your excellent work.

**Comment by fra589** [comment]
HI @jgalvarezr,  
  
Since I'm retired, summer is more convenient for journey than for computer programing 😃   
If backlash in always in my TODO list, it's a pretty complicated thing to implement and it's not a priority over other features that I intend to write next winter. So, I can't tell you that it will be available before sevral month.  
However, this feature should help solve mechanical backlash problems, not for inertia solving problem.  
If your CNC have a lot of inertia and fail at high speed, you need to adjust the acceleration parameters of your axis (parameters $120 to $125).  
I had similar problems when I started out, and fine-tuning maximum speeds and accelerations allowed me to solve them.  
  
@++;  
Gauthier.

**Comment by walterwissmann** [comment]
@jgalvarezr:
I can confirm Gauthier's arguments.

I would also like to mention:
When you have problems at high speed, you can also play with the current of your stepper drivers...

Bests
Walter

**Comment by jgalvarezr** [comment]
I don't know if it will be appropriate to continue this thread, but I need help to solve this problem. Accelerations work well in most cases. If I make a Y + movement and then an XY + the X motor does not accelerate, it tries to move instantly at the displacement speed, at that moment steps (all steps) skip. This occurs in some combinations of movements but not in all.

**Comment by fra589** [comment]
Hi @jgalvarezr,  
  
It will be better for all that you open a new thread for this specific case.  
Don't forget to post your config ($i, $$, etc.).  
A maximum of info is needed to save time.  
  
@++;  
Gauthier.

**Comment by davidlehrian** [comment]
I have implemented backlash on this code base as per https://github.com/shooter64738/grbl-mega. My changes are here https://github.com/davidlehrian/grbl-Mega-5X-backlash. If you are interested I can create a pull request and you can integrate these changes into your code base.

**Comment by fra589** [comment]
Hi @davidlehrian,  
  
Thanks for your proposal, backlash is still in my TODO list...
I'm traveling all the May, I'll watch this when I get back.  
  
@++;  
Gauthier.

**Comment by Matthias199** [comment]
@fra589 still on your todo list ?

**Comment by fra589** [comment]
Hi @Matthias199,  

Yes, it's still on the list, but I'm pretty busy on another projects...

@++;
Gauthier.

**Comment by davidlehrian** [comment]
Hello @Matthias199,
I have a branch of this code where backlash has been integrated and is available at https://github.com/davidlehrian/grbl-Mega-5X-backlash. I have been using it and it works great.

**Comment by Matthias199** [comment]
@davidlehrian Yeah that was the task for today, it works digital dial gauge says iam in 0.01mm backlash, what is for sure more an messure thing .... did you think of an auto backlash compensation with the endstops ?

**Comment by davidlehrian** [comment]
@Matthias199 There is no auto backlash compensation, it must be measured with a dial gauge and entered into the controller code via the commands:

X/Y/Z/A/B/C respectively

$140=x.xx
$141=x.xx
$142=x.xx
$143=x.xx
$144=x.xx
$145=x.xx

**Comment by Matthias199** [comment]
@davidlehrian Yeah i know that!
PLZ again the question was "did you think of an auto backlash compensation with the endstops"

This could help many people without tools to streamline the calibration.
Since voron uses klicki probes ... the regular endsops should be okey

**Comment by davidlehrian** [comment]
@Matthias199 No. A good dial indicator is an essential tool for all machinists in order to indicate the position of their workpiece. If they have a CNC mill and/or lathe, they have a dial indicator.
